File: frontend/src/graphql/evidence.ts
import { gql } from "@apollo/client";

export const GET_EVIDENCES = gql`
  query GetEvidences {
    getEvidences {
      id
      name
      evidence_type
      category
      description
      tags
      notes
      chain_of_custody
      analysis
      created_at
      updated_at
    }
  }
`;

export const GET_EVIDENCE = gql`
  query GetEvidence($evidenceId: String!) {
    getEvidence(evidenceId: $evidenceId) {
      id
      name
      evidence_type
      category
      description
      tags
      notes
      chain_of_custody
      analysis
      created_at
      updated_at
    }
  }
`;

export const CREATE_EVIDENCE = gql`
  mutation CreateEvidence($input: CreateEvidenceInput!) {
    createEvidence(input: $input) {
      id
      name
      evidence_type
      category
      description
      tags
      notes
      chain_of_custody
      analysis
      created_at
      updated_at
    }
  }
`;

export const UPDATE_EVIDENCE = gql`
  mutation UpdateEvidence($input: UpdateEvidenceInput!) {
    updateEvidence(input: $input) {
      id
      name
      evidence_type
      category
      description
      tags
      notes
      chain_of_custody
      analysis
      created_at
      updated_at
    }
  }
`;

export const DELETE_EVIDENCE = gql`
  mutation DeleteEvidence($evidenceId: String!) {
    deleteEvidence(evidenceId: $evidenceId) {
      success
      message
    }
  }
`;

File: backend/app/graphql/types/evidence.py
import strawberry
from typing import Optional, List
import uuid
from datetime import datetime


@strawberry.type
class EvidenceType:
    id: uuid.UUID
    name: str
    evidence_type: str
    category: str
    description: Optional[str]
    tags: Optional[str]
    notes: Optional[str]
    chain_of_custody: Optional[str]
    analysis: Optional[str]
    created_at: datetime
    updated_at: Optional[datetime]


@strawberry.input
class CreateEvidenceInput:
    name: str
    evidence_type: str
    category: str
    description: Optional[str] = None
    tags: Optional[str] = None
    notes: Optional[str] = None
    chain_of_custody: Optional[str] = None
    analysis: Optional[str] = None


@strawberry.input
class UpdateEvidenceInput:
    id: str
    name: Optional[str] = None
    evidence_type: Optional[str] = None
    category: Optional[str] = None
    description: Optional[str] = None
    tags: Optional[str] = None
    notes: Optional[str] = None
    chain_of_custody: Optional[str] = None
    analysis: Optional[str] = None
